actors/builtin/storage_market/storage_market_actor_state.go

```diff
@@ -1,93 +1,104 @@
 
 	// Total amount held in escrow, indexed by actor address (including both locked and unlocked amounts).
-	EscrowTable actor_util.BalanceTableHAMT
+	EscrowTable cid.Cid // BalanceTable
 
 	// Amount locked, indexed by actor address.
 	// Note: the amounts in this table do not affect the overall amount in escrow:
 	// only the _portion_ of the total escrow amount that is locked.
-	LockedReqTable actor_util.BalanceTableHAMT
+	LockedTable cid.Cid // BalanceTable
 
 	NextID abi.DealID
 
 	// Metadata cached for efficient iteration over deals.
-	CachedDealIDsByParty           DealsByParty
-	CachedExpirationsPending       DealExpirationQueue
-	CachedExpirationsNextProcEpoch abi.ChainEpoch
-	CurrEpochNumDealsPublished     int
-}
-
-func (st *StorageMarketActorState) UnmarshalCBOR(r io.Reader) error {
-	panic("replace with cbor-gen")
-}
-
-func (st *StorageMarketActorState) MarshalCBOR(w io.Writer) error {
-	panic("replace with cbor-gen")
+	DealIDsByParty map[string]DealMap // TODO: figure out a way to drop this
 }
 
 ////////////////////////////////////////////////////////////////////////////////
 // Deal state operations
 ////////////////////////////////////////////////////////////////////////////////
 
-func (st *StorageMarketActorState) _updatePendingDealStates(dealIDs []abi.DealID, epoch abi.ChainEpoch) (
-	amountSlashedTotal abi.TokenAmount) {
-
-	amountSlashedTotal = abi.NewTokenAmount(0)
+func (st *StorageMarketActorState) updatePendingDealStatesForParty(rt Runtime, addr addr.Address) (amountSlashedTotal abi.TokenAmount) {
+	// For consistency with OnEpochTickEnd, only process updates up to the end of the _previous_ epoch.
+	epoch := rt.CurrEpoch() - 1
 
-	for _, dealID := range dealIDs {
-		amountSlashedCurr := st._updatePendingDealState(dealID, epoch)
-		amountSlashedTotal = big.Add(amountSlashedTotal, amountSlashedCurr)
+	deals, ok := st.DealIDsByParty[addr.String()]
+	Assert(ok)
+	var extractedDealIDs []abi.DealID
+	for cachedDealID := range deals.M {
+		id, err := strconv.ParseUint(cachedDealID, 10, 64)
+		AssertNoError(err)
+		extractedDealIDs = append(extractedDealIDs, abi.DealID(id))
 	}
 
+	amountSlashedTotal = st.updatePendingDealStates(rt, extractedDealIDs, epoch)
 	return
 }
 
-func (st *StorageMarketActorState) _updatePendingDealState(dealID abi.DealID, epoch abi.ChainEpoch) (
-	amountSlashed abi.TokenAmount) {
+func (st *StorageMarketActorState) updatePendingDealStates(rt Runtime, dealIDs []abi.DealID, epoch abi.ChainEpoch) abi.TokenAmount {
+	amountSlashedTotal := abi.NewTokenAmount(0)
 
+	for _, dealID := range dealIDs {
+		amountSlashedTotal = big.Add(amountSlashedTotal, st.updatePendingDealState(rt, dealID, epoch))
+	}
+
+	return amountSlashedTotal
+}
+
+// TODO: This does waaaay too many redundant hamt reads
+func (st *StorageMarketActorState) updatePendingDealState(rt Runtime, dealID abi.DealID, epoch abi.ChainEpoch) (amountSlashed abi.TokenAmount) {
```

Comment:
Unlike much of the code inherited here, this one looks like it's doing too much real application logic in the state methods. Consider re-arranging this later so the important logic is in the actor itself.